### PR TITLE
Rework the remote topics to a tutorial-like flow

### PR DESCRIPTION
**Pull request by MelHopper**
Signed-off-by: MelHopper <melanieh@uk.ibm.com>

### CODE REVIEW

**Comment by MelHopper** [comment]
remote-overview, remotedeploy-single, reworked so far, remotedeploy-combo topic in progress

File: docs/_documentations/remote-jane-overview.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+description: Using Codewind remotely
+keywords: users, projects, Kubernetes, LDAP, user management, access management, login, deployment, pod, security, securing cloud connection, remote deployment of Codewind
+duration: 5 minutes
+permalink: remote-jane-overview
+type: document
+---
+
+# Using Codewind remotely
+
+Codewind can be used in one of 3 ways - [locally], [hosted] as an application on a cloud, or remotely. By using Codewind remotely, you can develop your code locally, but build and run your application in the cloud. Remote use of Codewind frees up your desktop resources, using the cloud's resources to build and run applications.
```
Comment (micgibso) [inline docs/_documentations/remote-jane-overview.md]: 3 ways => three ways

File: docs/_documentations/remote-jane-overview.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+After you install your local IDE and configure Codewind for local use, you will follow these steps to deploy Codewind to the cloud:
+
+1. Configure your IDE to connect to Codewind in the cloud
+2. Register all necessary security parameters
+
+Once you have connected your local Codewind to Codewind in the cloud, you will learn how to:
+
+1. create a new project that builds and runs in the cloud
+2. build and run an existing project in the cloud
+
```
Comment (micgibso) [inline docs/_documentations/remote-jane-overview.md]: Uppercase start to numbered sentence:
1. Create...
2. Build...

File: docs/_documentations/remote-overview.md [hunk 3]
```diff
@@ -10,36 +10,59 @@ type: document
 
 # Deploying Codewind remotely
 
-Deploy Codewind remotely to develop your code locally and build and run it remotely and securely in the cloud. This option is useful because it frees up resource on your laptop and uses the cloud's resources to build and run your apps.
+Codewind can be used in one of 3 ways - [locally], [hosted] as an application on a cloud, or remotely. By deploying Codewind remotely, developers can develop their code locally, but build and run your application in the cloud. Remote use of Codewind frees up desktop resources, using the cloud's resources to build and run applications.
```
Comment (micgibso) [inline docs/_documentations/remote-overview.md]: 3 ways => three ways

File: docs/_documentations/remote-overview.md [hunk 4]
```diff
@@ -10,36 +10,59 @@ type: document
 
 # Deploying Codewind remotely
 
-Deploy Codewind remotely to develop your code locally and build and run it remotely and securely in the cloud. This option is useful because it frees up resource on your laptop and uses the cloud's resources to build and run your apps.
+Codewind can be used in one of 3 ways - [locally], [hosted] as an application on a cloud, or remotely. By deploying Codewind remotely, developers can develop their code locally, but build and run your application in the cloud. Remote use of Codewind frees up desktop resources, using the cloud's resources to build and run applications. 
 
-After you install your IDE and configure Codewind for local use, follow these steps to deploy to the cloud:
+To learn how to use Codewind once it has been deployed remotely, see [Using Codewind remotely ](remote-jane-overview).
```
Comment (micgibso) [inline docs/_documentations/remote-overview.md]: [Using Codewind remotely ] <= delete space before ]

File: docs/_documentations/remote-overview.md [hunk 5]
```diff
@@ -10,36 +10,59 @@ type: document
...
+You will learn how to deploy Codewind to be used remotely. 
 
-Codewind remote services contain many different components. Use the Codewind CLI `cwctl` to install all the required components at once, or each component separately. 
+After you install your local IDE and configure Codewind for local use, you will use the Codewind CLI to:
 
-If you are running a pilot or a short lived demonstration environment with a single user, and are not using IBM Public Cloud, you can install everything into a single namespace and have one authentication server per Codewind instance. By deploying Keycloak and Codewind in a single namespace the deployment process is shorter and the removal process is simpler. 
+1. deploy Codewind to your cloud
+2. create additional instances of Codewind on your cloud, as required
+
+Finally, you will learn how to remove a remote deployment of Codewind.
```
Comment (micgibso) [inline docs/_documentations/remote-overview.md]: U/c start to numbered sentence:
1. Deploy...
2. Create...

File: docs/_documentations/remote-overview.md [hunk 6]
```diff
@@ -10,36 +10,59 @@ type: document
...
-- The installer can install Keycloak on its own, Keycloak and Codewind together, or Codewind pointing at an existing Keycloak. If you decide you want to deploy Keycloak on its own first, install it using the `cwctl install remote \` commands described in the following installation procedures and add the `--konly` flag. Then install a Codewind instance that uses Keycloak by adding the `--kurl` flag, which is the ingress of the Keycloak service.
-- Have an active Kubernetes context that points to your cluster. Codewind can run in OpenShift 3.11, OpenShift 4.3, OpenShift in IBM Public Cloud, standalone Kubernetes, and Kubernetes in Docker.
-- Have access to a keyring, a software application designed to store security credentials, such as usernames, passwords, and keys, together with a small amount of relevant metadata. Examples of a keyring are Keychain on macOS, Credential Manager on Windows, and Secret Service on Linux.
+Before deploying Codewind to the cloud, you must:
+
```
Comment (micgibso) [inline docs/_documentations/remote-overview.md]: Delete extra line.

File: docs/_documentations/remote-overview.md [hunk 7]
```diff
@@ -10,36 +10,59 @@ type: document
...
+2. **Ensure that you have access to the Codewind CLI `cwctl`.** To access the Codewind CLI, open a terminal window and navigate to the following hidden folder: `~/.codewind/<version>`. On Windows, navigate to the following folder: `%SystemDrive%\Users\<username>\.codewind\<version>`.
+
+3. **Have an active Kubernetes context that points to your cluster.** Codewind can run in OpenShift 3.11, OpenShift 4.3, OpenShift in IBM Public Cloud, standalone Kubernetes, and Kubernetes in Docker.
+
+4. **Have access to a keyring** A keyring is a software application designed to store security credentials, such as usernames, passwords, and keys, together with a small amount of relevant metadata. Examples of a keyring are Keychain on macOS, Credential Manager on Windows, and Secret Service on Linux.
+
+## Planning your remote deployment - Codewind and Authentication Services
 
-## Install Codewind and the authentication services
+A Keycloak must be used as the authentication service when using Codewind remotely. To use Codewind with Keycloak, you can either
```
Comment (micgibso) [inline docs/_documentations/remote-overview.md]: Add colon to end of sentence: ...either:

File: docs/_documentations/remote-overview.md [hunk 8]
```diff
@@ -10,36 +10,59 @@ type: document
...
+1. point Codewind to an existing Keycloak and then install Codewind remotely using this keycloak.
+2. use the Codewind CWCTL command on your local Codewind instance to install both Codewind remotely and Keycloak simultaneously.
+3. use the Codewind CWCTL command on your local Codewind instance to install Keycloak on it's own. Then, install a Codewind instance that uses your new Keycloak.
```
Comment (micgibso) [inline docs/_documentations/remote-overview.md]: Initial caps on numbered lists.

Comment (micgibso) [reply docs/_documentations/remote-overview.md]: Step 3. it's => its

File: docs/_documentations/remote-overview.md [hunk 9]
```diff
@@ -10,36 +10,59 @@ type: document
...
+
+- If you are running a pilot or a short lived demonstration environment with a single user, and are not using IBM Public Cloud, you can use the simplest deployment and removal process of deploying Keycloak and Codewind in a single namespace, by using the Codewind CWCTL command to install both Codewind remotely and Keycloak simultaneously. With this configuration, there is one authentication server per Codewind instance. 
+
+- If you are running in production, the best approach to simplify user administration is to install a single Keycloak service that is shared by many Codewind services. For this case, you need to install the authentication services separately from installing Codewind remotely. You will use a single keycloak to install multiple codewinds remotely.
+
+### Next Steps
+
+Once you have decided how you will be administering Codewind remotely, you can choose which installation procedure you need to follow:
+
+- to install both the remote Codewind instance and Keycloak simultaneously, see [deploying all-in-one](./remotedeploy-combo.html)
```
Comment (micgibso) [inline docs/_documentations/remote-overview.md]: Initial caps

File: docs/_documentations/remote-overview.md [hunk 10]
```diff
@@ -10,36 +10,59 @@ type: document
...
+
+Or:
 
-- [Install both Codewind and authentication services together](./remotedeploy-combo.html)
+- to install the remote Codewind instance and Keycloak individually, see [deploying components individually](./remotedeploy-single.html)
```
Comment (micgibso) [inline docs/_documentations/remote-overview.md]: Initial caps

File: docs/_documentations/remote-overview.md [hunk 11]
```diff
@@ -10,36 +10,59 @@ type: document
...
 
 Or:
 
-- [Install Codewind components and authentication services individually](./remotedeploy-single.html)
+- to install the remote Codewind instance only and pointing to an existing Keycloak, see [deploying with an existing keycloak](./remotedeploy-existingkeycloak.html)
```
Comment (micgibso) [inline docs/_documentations/remote-overview.md]: Initial caps

File: docs/_documentations/remotedeploy-existingkeycloak.md [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+
+Codewind includes a CLI to simplify the installation process. You can find the `cwctl` CLI in your HOME directory under the `~/.codewind/{version}` path.
+
+You will use the following command to install a new remote Codewind service:
+
+`cwctl --insecure install remote` 
+
+This command requires various flags to specify where and what to install which will be explained below.
+
+# Pre-requisites *NEEDS CHANGING FOR THIS SCENARIO?*
```
Comment (micgibso) [inline docs/_documentations/remotedeploy-existingkeycloak.md]: Resolve this question

File: docs/_documentations/remotedeploy-projects.md [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+duration: 5 minutes
+permalink: remotedeploy-projects
+type: document
+---
+
+# Objectives
+Now that you have configured Codewind to be used remotely, you are now ready to:
+
+1. create a new project
+2. move a project from your local Codewind instance to your remote Codewind instance on the cloud
```
Comment (micgibso) [inline docs/_documentations/remotedeploy-projects.md]: Initial caps

File: docs/_documentations/remotedeploy-single.md [hunk 14]
```diff
@@ -10,25 +10,53 @@ parent: installoncloud
...
 
-For OpenShift, Codewind is required to run as privileged and as root because it builds container images. Run the following commands where `<namespace>` is the namespace into which you plan to install Codewind:
-    - For Codewind on OpenShift, you must create the namespace first. To do this, enter: `oc create namespace <Codewind-namespace>`.
-    - If you are on IBM Public Cloud, you must install Keycloak and Codewind in separate namesapaces, for example, `oc create namespace <keycloak-namespace>` and then `oc create namespace <Codewind-namespace>`.
-    - For both IBM Public Cloud and OpenShift, run the following commands on only the Codewind namespace:
+- For IBM Public Cloud, you must install Keycloak and your remote Codewind in separate namespaces. 
+    Run the following commands where `<namespace>` is the namespace into which you plan to install Codewind:
+    1. Create the 2 separate namespaces, for example, `oc create namespace <keycloak-namespace>` and then `oc create namespace <Codewind-namespace>`.
+    2. run the following commands on only the Codewind namespace:
```
Comment (micgibso) [inline docs/_documentations/remotedeploy-single.md]: run => Run

File: docs/_documentations/remotedeploy-single.md [hunk 15]
```diff
@@ -150,8 +178,22 @@ INFO[0159] Codewind is available at: https://codewind-gatekeeper-k55333j0.myclus
...
+*WIP* You will now use the same keycloak to install multiple codewinds remotely.
+
+Additional remote Codewind instances can be generated against the existing keycloak by re-runnning the `./cwctl --insecure install remote \` command detailed in the section [Deploy remote Codewind services].. Each time you re-run the command, a new remote Codewind service is generated against the existing Keycloak service. You must have your own remote Codewind instance in order to generate additional remote codewind services.
+
+*SWAP THESE PARAGRAPHS AROUND?*
+Each user can have several remote Codewind instances allocated to them. This can be achieved by either:
+1. re-running the `./cwctl --insecure install remote \` command
+or
+2. assigning users through role-based acccess control. This is/*can be?* set up automatically using/*by?* the `cwctl` command *MORE DETAILS ARE WHERE?*
+
```
Comment (micgibso) [inline docs/_documentations/remotedeploy-single.md]: Initial caps: 
re-running => Re-running
assigning => Assigning...

File: docs/_documentations/remotedeploy-vscode.md [hunk 16]
```diff
@@ -10,7 +10,20 @@ parent: installoncloud
...
+# Connecting VS Code to your remote Codewind instance
+
+Ensure you have [satisfied all prequisites](./remote-deploy-jane.html). 
+
+
+# Objectives
+
+In this topic you will:
+
+1. connect Codewind on VS Code to the remote instance of Codewind in the cloud.
```
Comment (micgibso) [inline docs/_documentations/remotedeploy-vscode.md]: connect => Connect

File: docs/_documentations/remotedeploy-projects.md (hunk 13 above)
Comment (MelHopper) [reply docs/_documentations/remotedeploy-projects.md]: note this topic renamed to 'remotedeploy-projects-vscode.md

**Comment by tobespc** [comment]
Force merging as @MelHopper has signed the eclipse agreement but just omitted the flag from this PR

**Pull request merged**